apps/trading/components/market-banner/market-termination-banner.spec.tsx

```diff
@@ -0,0 +1,98 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import * as Types from '@vegaprotocol/types';
+import { MemoryRouter } from 'react-router-dom';
+import type { MockedResponse } from '@apollo/client/testing';
+import { MockedProvider } from '@apollo/client/testing';
+import type { MarketViewProposalsQuery } from '@vegaprotocol/proposals';
+import { MarketViewProposalsDocument } from '@vegaprotocol/proposals';
+import { MarketTerminationBanner } from './market-termination-banner';
+
+const proposalMock: MockedResponse<MarketViewProposalsQuery> = {
+  request: {
+    query: MarketViewProposalsDocument,
+    variables: { inState: Types.ProposalState.STATE_PASSED },
+  },
+  result: {
+    data: {
+      proposalsConnection: {
+        edges: [
+          {
+            node: {
+              id: 'first-id',
+              state: Types.ProposalState.STATE_PASSED,
+              terms: {
+                closingDatetime: '2023-09-27T11:48:18Z',
+                enactmentDatetime: '2023-09-30T11:48:18',
+                change: {
+                  __typename: 'UpdateMarketState',
+                  updateType: '',
```

Comment:
Type cast on line 69 is risky as it will make the test brittle. If you modify the query above and have an incorrect or missing field the test will break.

Instead remove the type cast and use the correct value for `updateType`

```suggestion
                  updateType: MarketUpdateType.MARKET_STATE_UPDATE_TYPE_TERMINATE,
```

Reply:
filled up, thanks